idseq_pipeline/commands/common.py

```diff
@@ -211,6 +211,43 @@ def __exit__(self, *args):
             self.tail_subproc.kill()
 
 
+class AsyncHandler:
+    def __init__(self):
+        self.threads = []
+        pass
+
+    def launch(self, target, args):
+        if type(args) == str:
+            args = (args, )
+        t = threading.Thread(target=target, args=args)
+        self.threads.append(t)
+        t.start()
+
+    def wait_on_all(self):
+        write_to_log("Waiting on AsyncHandler threads...")
+        for t in self.threads:
+            t.join()
+        write_to_log("AsyncHandler threads finished.")
+
+    def launch_aws_cp(self, src, dst):
```

Comment:
I would prefer if you call this `lauch_aws_upload()` to indicate we are only really using it for uploads.